vnext/JSI/Shared/ChakraApi.h

```diff
@@ -0,0 +1,465 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef CHAKRACORE
+#include "ChakraCore.h"
+#else
+#ifndef USE_EDGEMODE_JSRT
+#define USE_EDGEMODE_JSRT
+#endif
+#include <jsrt.h>
+#endif
+
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <string_view>
+
+// We use macros to report errors.
+// Macros provide more flexibility to show assert and provide failure context.
+
+// Check condition and crash process if it fails.
+#define ChakraVerifyElseCrash(condition, message)          \
+  do {                                                     \
+    if (!(condition)) {                                    \
+      assert(false && "Failed: " #condition && (message)); \
```

Comment:
>&& [](start = 19, length = 2)

For my understanding: What does the '&&' represent here?

Reply:
This is a common way to provide a message for the assert macro. It accepts only boolean and && allows to add a message because the string is always true. The assert then show the whole text given to it.

---
In reply to: [470172679](https://github.com/microsoft/react-native-windows/pull/5710#discussion_r470172679) [](ancestors = 470172679)